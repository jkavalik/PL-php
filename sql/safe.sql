begin;
--
-- Test forbidden functions
--

CREATE FUNCTION test_show_source() RETURNS text
LANGUAGE plphp AS $$
	return show_source();
$$;
SELECT test_show_source();
SELECT * FROM test_show_source();

CREATE FUNCTION test_exec() RETURNS text
LANGUAGE plphp AS $$
	return exec("psql --version");
$$;
SELECT test_exec();
SELECT * FROM test_exec();


rollback;