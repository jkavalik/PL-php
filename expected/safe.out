begin;
BEGIN
--
-- Test forbidden functions
--
CREATE FUNCTION test_show_source() RETURNS text
LANGUAGE plphp AS $$
	return show_source();
$$;
CREATE FUNCTION
SELECT test_show_source();
 test_show_source 
------------------
 
(1 row)

SELECT * FROM test_show_source();
 test_show_source 
------------------
 
(1 row)

CREATE FUNCTION test_exec() RETURNS text
LANGUAGE plphp AS $$
	return exec("psql --version");
$$;
CREATE FUNCTION
SELECT test_exec();
 test_exec 
-----------
 
(1 row)

SELECT * FROM test_exec();
 test_exec 
-----------
 
(1 row)

rollback;
ROLLBACK
